api/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -87,6 +87,11 @@ type OpenTelemetryCollectorSpec struct {
 	// +optional
 	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
 	Resources v1.ResourceRequirements `json:"resources,omitempty"`
+
+	// Toleration to schedule OpenTelemetry Collector pods.
```

Comment:
Please, document under which modes this is relevant: `daemonset` and `deployment`. How about statefulsets? Similarly, you might need to add some validation, so that we don't accept tolerations for unsupported modes (sidecar comes to my mind, for instance).

Reply:
This is relevant in `daemonsets`, `deployments` & `statefulsets` (I didn't know this existed). I thought to add validation but noticed we do not have validation in place for example `replicas` are relevant only to `deployments` & `statefulsets` but not `daemonsets` and `sidecars`. Currently, the CRD accepts replicas for unsupported modes too. 

Could you point me to where exactly do you think CRD validation makes sense in the operator flow?

Reply:
Good catch about the replica size. Do you mind creating an issue for it? The appropriate place for such a validation would be here: 

https://github.com/open-telemetry/opentelemetry-operator/blob/4ead5fe1079b6d83c5c1796ee45ef93c6f88671f/api/v1alpha1/opentelemetrycollector_webhook.go#L59-L81

Reply:
Added the replicas validation in my latest commit.